File: src/main/java/cn/misection/relation/dao/RelationDAO.java
package cn.misection.relation.dao;

import java.util.List;

import com.alibaba.excel.EasyExcel;
import com.sun.istack.internal.NotNull;

import cn.hutool.json.JSONUtil;
import cn.hutool.log.Log;
import cn.misection.relation.callback.RelationListener;
import cn.misection.relation.entity.PersonRelationPair;

/**
 * @author Military Intelligence 6 root
 * @version 1.0.0
 * @ClassName RelationReader
 * @Description TODO
 * @CreateTime 2022年02月24日 23:50:00
 */
public class RelationDAO {

  @NotNull
  public static List<PersonRelationPair> readExcel(String path) {
    final List<PersonRelationPair>[] result = new List[1];
    EasyExcel.read(path, PersonRelationPair.class, new RelationListener(
        (List<PersonRelationPair> dataList) -> {
          for (int i = 0; i < dataList.size(); i++) {
            Log.get().info("读取到第 {} 条数据 {}", i, JSONUtil.toJsonStr(dataList.get(i)));
          }
          result[0] = dataList;
        })).doReadAll();
    return result[0];
  }
}

File: KtMapper/src/main/kotlin/cn/misection/relation/dao/RelationDAO.kt
package cn.misection.relation.dao

import cn.hutool.json.JSONUtil
import cn.hutool.log.Log
import cn.misection.relation.callback.RelationListener
import cn.misection.relation.entity.PersonRelationPair
import com.alibaba.excel.EasyExcel

/**
 * @ClassName RelationDAO
 * @author Military Intelligence 6 root
 * @version 1.0.0
 * @Description TODO
 * @CreateTime 2022年02月26日 12:07:00
 */
object RelationDAO {

    fun readExcel(path: String): List<PersonRelationPair> {
        var result: List<PersonRelationPair> = ArrayList()
        EasyExcel.read(path, PersonRelationPair::class.java, RelationListener { dataList: List<PersonRelationPair> ->
            run {
                for ((index: Int, data: PersonRelationPair) in dataList.withIndex()) {
                    Log.get().info("读取到第 {} 条数据 {}", index, JSONUtil.toJsonStr(data))
                }
            }
            result = dataList
        })
        return result
    }
}
